docs/development/vscode.md

```diff
@@ -0,0 +1,143 @@
+# Setting Up VS Code for Debugging Zulip
+
+## Overview
+
+Set up VS Code to debug the django zulip sever. The following can only
+setup VS Code to debug the *django application* and not the *JS*
+(frontend) part of zulip. This is still a work in progress and any
+suggestions to simplify or ease the workflow are encouraged.
+
+I hope this helps the community of VS Code users working on zulip and
+helps them significantly ! :D
+
+## Steps
+
+1. Download and install VS Code.
+2. Download and install the
+   [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+   extension for VS Code by Microsoft.  
+3. Set up the vagrant dev environment from [here](./setup-advanced.md)
+4. Make sure you run the dev server at least once. ( So that you know
+   everything is working before making changes to the files )
+5. Expose *port* in Vagrant file
```

Comment:
Can we do this via a setting in `~/.zulip-vagrant-config` , so no editing is required?  That's make it much more pleasant to debug.

Reply:
Yeah, I had done that in my env. Surely would be great to make it as easy as possible